webapp/src/components/cluster.tsx

```diff
@@ -0,0 +1,210 @@
+                <Button onClick={() => {
+                  setForm(undefined)
+                  setVisibleForm(true)
+                }}>{Locale.UI.Create}</Button>
+              </ButtonGroup>
+            </>} dataIndex="operate" key="operate" render={operateRender} />
+          )}
+        </Table>
+      </Card>
+      <Modal
+        title={form === undefined ? Locale.UI.Create : Locale.UI.Edit}
+        visible={visibleForm}
+        onOk={() => { formApi?.submitForm() }}
+        onCancel={() => { setVisibleForm(false) }}
+        centered
+        width={500}
+        height={500}
+        bodyStyle={{ overflow: 'auto' }}
+      >
+        <Form
+          labelPosition="left"
+          labelAlign="left"
+          labelWidth={150}
+          style={{ paddingRight: '20px' }}
+          onSubmit={(values) => {
+            if (form === undefined) {
+              backendSaveApi(values)
+                .then(data => {
+                  console.log(data);
+                  list();
+                  setVisibleForm(false);
+                }).catch(() => { });
+            } else {
+              backendUpdateApi(values)
+                .then(data => {
+                  console.log(data);
+                  list();
+                  setVisibleForm(false);
+                }).catch(() => { });
+            }
+          }}
+          getFormApi={setFormApi}
+        >
+          <Form.Input
+            field="name"
+            label="Name"
+            trigger='blur'
+            rules={[
+              { required: true, message: 'required error' },
+              { type: 'string', message: 'type error' },
+            ]}
+            disabled={form !== undefined}
+            initValue={form?.name}
+          />
+          <Form.Input
+            field="routingGroup"
+            label="RoutingGroup"
+            trigger='blur'
+            rules={[
+              { required: true, message: 'required error' },
+              { type: 'string', message: 'type error' },
+            ]}
+            initValue={form?.routingGroup}
+          />
+          <Form.Input
+            field="proxyTo"
+            label="ProxyTo"
+            trigger='blur'
+            rules={[
+              { required: true, message: 'required error' },
+              { type: 'string', message: 'type error' },
+            ]}
+            initValue={form?.proxyTo}
+          />
+          <Form.Input
+            field="externalUrl"
+            label="ExternalUrl"
+            trigger='blur'
+            rules={[
+              { required: true, message: 'required error' },
+              { type: 'string', message: 'type error' },
+            ]}
+            initValue={form?.externalUrl}
+          />
+          <Form.Switch label="Active" field='active' initValue={form?.active || false} />
+        </Form>
+      </Modal>
+    </>
+  );
+}
+
+const SwitchRender = (props: {
+  text: any;
+  record: any;
+  index: any;
```

Comment:
is it possible to get rid of these `any` types?

Reply:
👌 Okay, change it to a specific type.